WCFDataService/Client/System/Data/Services/Client/Binding/BindingUtils.cs

```diff
@@ -86,8 +86,9 @@ internal static void VerifyObserverNotPresent<T>(object oec, string sourceProper
 
             if (typedCollection.Observer != null)
             {
-                throw new InvalidOperationException(Strings.DataBinding_CollectionPropertySetterValueHasObserver(sourceProperty, sourceType));
```

Comment:
Do we need check whether to throw according to the added flags? Since this is a behavior change as well